### PR TITLE
Identify steps that can be moved to startup 

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

This PR is in preparation to #7514 

The following steps are moved to PartitionStartupStep
* StateCotrollerPartitionStep
* LogDeletionParitionStep

Now only their types have changes. These services are still re-installed during each transitions. This will be changed in future when we split all steps into startup vs transition steps. 

ZeebeDbPartitionStep can also be moved to PartitionStartupStep, this will be done in a different PR.

## Related issues

Related #7514

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java [hunk 1]
```diff
@@ -31,6 +31,8 @@
    */
   ZeebeDb openDb();
 
+  void closeDb() throws Exception;
```
Comment (saig0) [inline broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java:34]: :question:  What is the difference to the `close()` method? The state controller already implements `AutoCloseable`.

**Review by saig0 — COMMENTED** [review]
Looks good :+1: 

I've only a minor question. Please have a look at the failing smoke test.

File: broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java (hunk 1 above)
Comment (deepthidevaki) [reply broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java:34]: DB is opened by ZeebeDBPartitionStep. So it should be closed by the same. In future StateControllerStep will be only installed once at the startup, and closed only when the broker is closed. But we have to close/re-open zeebedb during role transitions.

Comment (saig0) [reply broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java:34]: Do you mean that `StateControllerImpl#closeDb()` and `StateControllerImpl#close()`will behave differently (i.e. it doesn't remove the runtime forlder)? 
Currently, `close()` delegates to `closeDb()`.

Comment (deepthidevaki) [reply broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java:34]: Both behave similar now. May be we don't need to close db in ``StateControllerImpl#close()`. We can assume that ZeebeDBStep will close it and remove the runtime folder.

Comment (saig0) [reply broker/src/main/java/io/camunda/zeebe/broker/system/partitions/StateController.java:34]: I'm okay if you see a need in it :+1:

**Review by saig0 — APPROVED** [review]
:+1:

**Pull request merged**